modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java

```diff
@@ -535,6 +557,26 @@ protected String getPermissionCheckerResourceName(Object id) {
 		return DLFileEntry.class.getName();
 	}
 
+	@Override
+	public void update(
+			Collection<Document> documents,
+			Map<String, Serializable> parameters)
+		throws Exception {
+
+		_batchDocuments = new HashMap<>();
```

Comment:
This shouldn't be an instance variable; think about what would happen if the resource received multiple batch requests at the same time. As long as all updates in a batch request are executed in the same thread, then you could use a thread local variable to communicate the different parts of the code.

Reply:
You can use [this thread local](https://github.com/liferay/liferay-portal/blob/69ed7688ad95a8c56a51f35307e3edbcf6393464/modules/apps/batch-engine/batch-engine-api/src/main/java/com/liferay/batch/engine/thread/local/BatchEngineThreadLocal.java#L15-L17).